installer/templates/phx_single/mix.exs

```diff
@@ -55,8 +55,8 @@ defmodule <%= @app_module %>.MixProject do
        app: false,
        compile: false,
        depth: 1},<% end %><%= if @mailer do %>
-      {:swoosh, "~> 1.5"},
-      {:finch, "~> 0.13"},<% end %>
+      {:swoosh, "~> 1.16"},
+      {:req, "~> 0.5.4"},<% end %>
```

Comment:
Just so you know, I'm considering some changes before Req v1.0 that are maybe worth mentioning here:

1. Since day 1, Req retries on errors by default. I've had some feedback over the years this is a bit surprising. There's a recently opened issue too: https://github.com/wojtekmach/req/issues/383. If you'd like to chime in, that's appreciated. I added some extra context in Swoosh too: https://github.com/swoosh/swoosh/pull/830/files#r1694016381.

2. Since day 1, Req brings in jason dependency. I'd like to drop it before v1.0 in favour of built-in :json from OTP 27. I'd be OK even conditionally vendoring it in for pre-27. This is tracked here: https://github.com/wojtekmach/req/issues/386.

Btw, [Finch is dropping `:castore` dependency](https://github.com/sneako/finch/pull/274) and once Finch is out with this change, I'm keen on requiring that Finch version as minimum in Req. I'd probably cut Req v0.6.0 for this change. This is effectively requiring OTP 25 (and running on system that have built-in CA) for everything to work out of the box.

`{:req, "~> 0.5.4"}` is definitely good for now. I'd be happy to send a patch bumping that to ~> 0.6/1.0 in the future.

Dunno how relevant are these for Phoenix but I thought Id mentioned it anyway!

Reply:
Both Swoosh and Phoenix depends on Jason, so making it optional on Req won't affect us. I also don't think OTP 25 requirement is an issue, this is for new projects and that's the minimum supported OTP version nowadays anyway.

Reply:
Regarding req retries by default and swoosh, this is all moot, sorry about the noise. In req the default is retry: :safe_transient, safe meaning only for GET/HEAD requests. Sending emails would be typically done using POST. So we are good.